backend-rust/src/indexer.rs

```diff
@@ -138,11 +138,14 @@ SELECT height FROM blocks ORDER BY height DESC LIMIT 1
         let (sender, receiver) = tokio::sync::mpsc::channel(self.config.max_processing_batch);
         let receiver = tokio_stream::wrappers::ReceiverStream::from(receiver)
             .ready_chunks(self.config.max_processing_batch);
-        let traverse_future = traverse_config.traverse(self.block_pre_processor, sender);
-        let process_future = processor_config.process_event_stream(self.block_processor, receiver);
+        let traverse_future =
+            tokio::spawn(traverse_config.traverse(self.block_pre_processor, sender));
+        let process_future =
+            tokio::spawn(processor_config.process_event_stream(self.block_processor, receiver));
         info!("Indexing from block height {}", self.start_height);
-        let (result, ()) = futures::join!(traverse_future, process_future);
-        Ok(result?)
+        let (traverse_result, process_result) = futures::join!(traverse_future, process_future);
+        process_result?;
+        Ok(traverse_result??)
```

Comment:
Using `try_join!` instead of `join!` might be more elegant here.

Reply:
The difference is that `join!` will wait for all of them to finish, but `try_join!` will return if just one of them results in `Err`. So this was my intent.

Reply:
Why does it matter that both of them finishes if an error occurs in either of them? Are you saying that the side effects of one of them needs to be observed even when the other one fails? A comment explaining why would be good.

Reply:
It is a choice, and either are valid approaches to me. In this case, if traverse exits with an error, we still complete the current processing before exiting.

Reply:
I definitely don't think this is just a meaningless distinction/choice. I would not say both are equally valid but that they should be used in different situations.

The fact that you do it like this suggests to the reader that the side effects matter. If the side effects matter, you definitely should write it like this and explain why the side effects matter in a comment. But if the side effects do not matter, then writing it like this is misleading, I feel. It makes it seem like the side effects matter, when they actually don't. Using `try_join!` would make it clear that the side effects do not matter. So I would not say they are equally valid, but that they should be used appropriately (`join!` when side effects matter, `try_join!` when they don't).

Also the error is obviously reported faster if you don't wait for the other task to finish, but I imagine the performance is not important here.

Reply:
I did not call the choice meaningless.

The traverse task will result in an error if it fails to prepare blocks for long enough, this could be due to network issues with the concordium-nodes, in anycase the traverser decided to shut down, so allowing the process_task to gracefully shutdown seems better.
In the end we can always recover, but gracefully shutting down will mean the batch of blocks that are currently getting processed will complete before exit and I don't see why the traverser should prevent that.